scripts/upgrade

```diff
@@ -90,49 +116,73 @@ ynh_script_progression --message="Upgrading NGINX web server configuration..." -
-ynh_script_progression --message="Making sure dedicated system user exists..."
+ynh_script_progression --message="Making sure dedicated system user exists..." --weight=1
 
 # Create a dedicated user (if not existing)
 ynh_system_user_create --username=$app
 
 #=================================================
 # PHP-FPM CONFIGURATION
 #=================================================
-ynh_script_progression --message="Upgrading PHP-FPM configuration..."
+ynh_script_progression --message="Upgrading PHP-FPM configuration..." --weight=2
 
 # Create a dedicated PHP-FPM config
-ynh_add_fpm_config --package="$extra_php_dependencies"
+ynh_add_fpm_config --usage=low --footprint=low --package="$extra_php_dependencies"
 
 #=================================================
-# SPECIFIC UPGRADE
+# PHP-FPM CONFIGURATION
 #=================================================
-# INSTALL AND INITIALIZE COMPOSER
+ynh_script_progression --message="Installing Composer..." --weight=1
+
+#ynh_secure_remove --file="$final_path/composer.lock"
+
+# Upgrade composer itself
+ynh_install_composer --phpversion=$phpversion --workdir=$final_path --install_args="--prefer-source --no-dev"
+
 #=================================================
-ynh_script_progression --message="Installing Composer..." --weight=10
+# SETUP APPLICATION CONFIG
+#=================================================
+ynh_script_progression --message="Building..."
 
-# Install Composer
-ynh_install_composer
+# Setup application config
+ynh_add_config --template="../conf/.env.example" --destination="$final_path/.env"
 
-ynh_exec_warn_less ynh_composer_exec --commands=\"dump-autoload\"
+# Setup custom user.css file
+#cp ../conf/user.css.example $final_path/public/dist/user.css
 
-( cd $final_path && php$phpversion artisan migrate -n --force )
-( cd $final_path && php$phpversion artisan config:clear -n )
-( cd $final_path && php$phpversion artisan config:cache -n )
+#=================================================
+# INSTALL AND INITIALIZE COMPOSER
+#=================================================
 
-ynh_backup_if_checksum_is_different --file="$final_path/.env"
-# Recalculate and store the checksum of the file for the next upgrade.
-ynh_store_file_checksum --file="$final_path/.env"
+if [ "$upgrade_type" == "UPGRADE_APP" ]
+then
+	ynh_script_progression --message="Upgrading Lychee source files..." --weight=4
+	pushd $final_path
+		composer install --no-dev
+		php$phpversion artisan key:generate --no-interaction --force
+		php$phpversion artisan migrate --no-interaction --force
+	popd
+fi
 
 #=================================================
 # SECURE FILES AND DIRECTORIES
 #=================================================
 
-# Set permissions on app files
+# Set permissions to app files
 chown -R $app: $final_path
-#chmod -R 750 /home/yunohost.$app/{uploads,data}
+chmod -R 750 $final_path/bootstrap/cache
+chown -R $app: $public_path
+chmod -R 750 $public_path
```

Comment:
```suggestion
chmod 750 $final_path/bootstrap/cache
chown -R $app: $public_path
chmod 750 $public_path
```

I tend to avoid recursively setting such chmods because setting every regular file as executable is weird.
Usually the intention is just to protect the "entrance" of the folder from random users